### PR TITLE
Scheduling tab

### PR DESCRIPTION
**Pull request by rpiaggio**
Kapture 2023-05-31 at 01 25 55

This PR depends on a local version of core which has needed changes to the timing windows, but doesn't have the new sequence model.

Obviously, merging this will break the build. Not sure how to handle this. One way to proceed is to merge this and immediately start working on the sequence model migration. Explore will not build until we complete that. Another is to work on the sequence model migration first and then rebase this PR on top of that.

### CODE REVIEW

**Comment by toddburnside** [comment]
> Obviously, merging this will break the build. Not sure how to handle this. One way to proceed is to merge this and immediately start working on the sequence model migration. Explore will not build until we complete that. Another is to work on the sequence model migration first and then rebase this PR on top of that.

I'm hesitant to break the build and create a logjam of PRs. I think the latter option is better.

File: explore/src/main/scala/explore/observationtree/SchedulingGroupObsList.scala [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    <.span(twt match
+      case TimingWindowInclusion.Include => ExploreStyles.TimingWindowInclude
+      case TimingWindowInclusion.Exclude => ExploreStyles.TimingWindowExclude
+    )(Icons.Circle)
+  )
+
+  private given Render[Option[TimingWindowEnd]] = Render.by {
+    case None                                    => Icons.ArrowRight
+    case Some(TimingWindowEnd.At(_))             => Icons.ArrowRightToLine
+    case Some(TimingWindowEnd.After(_, None))    => Icons.ArrowRightToLine
```
Comment (toddburnside) [inline explore/src/main/scala/explore/observationtree/SchedulingGroupObsList.scala:75]: Did you want the same icon for both?

**Review by toddburnside — APPROVED** [review]
LGTM - I like the use of the icons in the group headers

**Comment by cquiroz** [comment]
It seems we need to prepare for a set of PR related to shane changes on core/odb related to timing. Maybe we do that first?

File: explore/src/main/scala/explore/tabs/ConstraintsTabContents.scala [hunk 2]
```diff
@@ -43,6 +43,7 @@ import explore.optics.all.*
 import explore.shortcuts.*
 import explore.shortcuts.given
 import explore.syntax.ui.*
+import explore.tabs.ConstraintsTabContents
```
[the remark this answers is not in the document]
Comment (rpiaggio) [reply explore/src/main/scala/explore/tabs/ConstraintsTabContents.scala:46]: Good catch, will remove.

**Pull request merged**